### PR TITLE
Look for local .cabal files when generating fpco/stackage-server#214

### PR DESCRIPTION
**Pull request by snoyberg**

### CODE REVIEW

File: src/Action/Generate.hs [hunk 1]
```diff
@@ -126,17 +128,25 @@ readHaskellOnline timing settings download = do
     return (cbl, want, source)
 
 
-readHaskellDirs :: Timing -> [FilePath] -> IO (Map.Map String Package, Set.Set String, Source IO (String, URL, LStr))
-readHaskellDirs timing dirs = do
+readHaskellDirs :: Timing -> Settings -> [FilePath] -> IO (Map.Map String Package, Set.Set String, Source IO (String, URL, LStr))
+readHaskellDirs timing settings dirs = do
     packages <- map (takeBaseName &&& id) . filter ((==) ".txt" . takeExtension) <$> concatMapM listFilesRecursive dirs
+    cabals <- concatMapM listFilesRecursive dirs >>=
```
Comment (ndmitchell) [inline src/Action/Generate.hs]: The listFilesRecursive should be done once and shared breed packages and cabals, since it can be relatively expensive.

**Comment by snoyberg** [comment]
Good call

File: src/Action/Generate.hs (hunk 1 above)
Comment (snoyberg) [inline src/Action/Generate.hs]: OK, change pushed.

**Comment by ndmitchell** [comment]
Thanks for those changes, it all looks great. I added a few minor tweaks on top, but nothing major. Let me know if you want a release, or if running off a git hash is equally easy for you.

**Pull request merged**